### ISSUE TITLE
❓ Understanding smoothed CrossEntropyLoss

### ISSUE THREAD

**Issue by Colanim** [pytorch/fairseq#2273]
## ❓ Questions and Help

I'm trying to understand the label-smoothed cross-entropy loss of Fairseq :
https://github.com/pytorch/fairseq/blob/d2b5265a60c4016d9fc3cbd55ad983dd86f1aa6f/fairseq/criterions/label_smoothed_cross_entropy.py#L12-L29

I compare the output of this loss with the output of TF2 smoothed cross-entropy loss ([CategoricalCrossentropy](https://www.tensorflow.org/api_docs/python/tf/keras/losses/CategoricalCrossentropy)) in [this Colab notebook](https://colab.research.google.com/drive/1AoD75f4heMvBQKCFY34EGfzXTdPTAq0k?usp=sharing), but the results are differents...

**Anyone know why ?**

**Comment by shamanez** [comment]
Did you use the same epsilon values?

**Comment by Colanim** [comment]
Yes I same input and same parameters for both, as shown in the Colab notebook.
I don't think I did any mistake. I'm looking for a TF equivalent of the fairseq loss.

**Comment by Colanim** [comment]
I found my mistake : I was comparing CrossEntropyLoss with NLLLoss...  
Also, loss is summed in Fairseq.

By adding a `log_softmax`, it solves the problem.

Here is [the Colab notebook with the fixed code.](https://colab.research.google.com/drive/1vLXRnY__P9KhHW2MtUrMECRI4qFRuSx6?usp=sharing)
